products/BellHybrid/services/db/agents/QuotesAgent.cpp

```diff
@@ -30,9 +35,12 @@ namespace Quotes
         const auto request = std::dynamic_pointer_cast<Messages::ReadRandomizedQuoteRequest>(query);
         if (request) {
             const auto id = shuffleQuoteModel.getId();
+            const auto quotesGroup =
+                settings->getValue(settings::Quotes::selectedGroup, settings::SettingsScope::Global);
             LOG_DEBUG("Shuffle id: %d", id);
-            const auto result = quotesDB->query(Queries::readQuote, id);
-            if (result->getRowCount() > 0) {
+            auto readQuery    = quotesGroup == customGroup ? Queries::readCustomQuote : Queries::readPredefinedQuote;
+            const auto result = quotesDB->query(std::move(readQuery), id);
```

Comment:
```suggestion
            const auto result = quotesDB->query(readQuery, id);
```

No need to move here, as we're dealing with `const char *`, which is trivially copyable. Also, `readQuery` can be defined `const` then

Reply:
done